DOC-673: Rework shortest path query (#752)
Macrometacorp/docs: docs/graphs/graph-queries/shortest-path.md

---
sidebar_position: 20
title: Shortest Path Query
---

> A _shortest path query_ is a query that finds the shortest path between two given documents in your graph. The two documents are referred to as the `startVertex` and `targetVertex`. When you execute this query, you receive a result in the form of a set with two items for each vertex on the shortest path:

Reviewer: When you execute this query, you receive a result as an array where each element represents a vertex in the shortest path and consists of two items:

(otherwise it's not clear what the output format is)

Author: Thanks, Martin. BTW, you can make a suggestion directly by clicking the icon with the plus and minus inside a page outline. It's generally easier for us to understand that way.

Author: @martin-macrometa, I made the change, but I'm not sure it makes sense. How can "each element represents a vertex in the shortest path" and the results being an edge and the vertex both be true?

Reviewer: @diana-macrometa It's a nested data structure. it's something like this:

vertex0, edge0
vertex1, edge1
vertex2, edge2
...

It looks okay to me now. One typo - "two item" should be "two items"

> A _shortest path query_ is a query that finds the shortest path between two given documents in your graph. The two documents are referred to as the `startVertex` and `targetVertex`. When you execute this query, you receive the result as an array where each element represents a vertex in the shortest path and consists of two item:

Reviewer 2: ```suggestion
A _shortest path query_ is a query that finds the shortest path between two given documents in your graph. The two documents are referred to as the `startVertex` and `targetVertex`. When you run this query, you receive the result as an array where each element represents a vertex in the shortest path and consists of two item:
```

- The vertex itself.
- The edge pointing to it.

Shortest path queries enable you to quickly find the shortest path between two vertices in your graph, allowing you to traverse and analyze your graph's structure. This can be particularly useful in applications where finding the most efficient path is critical, such as route planning or supply chain optimization.

## Shortest Path Query in Action

> Let's take a look at a simple example to explain how it works. In this high-level example, we have a graph with four vertices labeled A, B, C, and D. The edges are labeled with their weight, or length. The graph looks like this

Reviewer 2: ```suggestion
In this high-level example, you have a graph with four vertices labeled A, B, C, and D. The edges are labeled with their weight, or length. The graph looks like this:
```

Not sure first sentence is necessary

![traversal graph](/img/traversal_graph.png)

> We will use the following parameters for our query:

Reviewer 2: ```suggestion
Use the following parameters for your query:
```

- Start at the vertex **A**.
- Finish with the vertex **D**.

> In this example, the shortest path between **A** and **D** is through vertices A, B, C and D in that order. The shortest path statement will return the following pairs:

Reviewer 2: ```suggestion
In this example, the shortest path between **A** and **D** is through vertices A, B, C and D in that order. The shortest path statement returns the following pairs:
```
